### DISCUSSION TITLE
cloudcontroller and providerid after creation

### DISCUSSION THREAD

**Discussion by chris93111** [k3s-io/k3s#10392] [General]
hello i have a k3s cluster running since many years and is initialize with default cloudcontroller, but now i would like use vsphere csi and need ccm/cpi with providerid of vsphere

they are any solution except recreate a cluster for disable cloudcontroller ? 

thanks

**Comment by brandond** [comment]
As far as I know those fields are immutable after the node joins the cluster. You could try deleting and rejoining the node?

**Reply by chris93111** [reply]
@brandond thanks for your reply, i can delete the node and join but can i do with control plane ? my cluster have 3 controlplane and 9 worker

**Reply by brandond** [reply]
You'd need to do the same for all the nodes, regardless of whether they are agent or server.

**Comment by chris93111** [comment]
Hi is worker without change provider-id thanks
